modules/rgbd/doc/kinfu_icp.markdown

```diff
@@ -0,0 +1,111 @@
+# ICP algorithm in Kinect Fusion {#kinfu_icp}
+
+The Iterative closest point (ICP) function minimizes the PointToPlane Distance (PPD) between the corresponding points in two clouds of points and normals.
+Specifically, it is the distance from the point ***P*** to the plane with the normal ***N*** in which the point ***Q*** located
+
+The main equetion, which it needs to minimize:
+
+<img src="https://render.githubusercontent.com/render/math?math=E=\sum\left\|ppd(p_{i},q_{i},n_{i})\right\|_{2}\rightarrow0">
+
+
+Let's watch what is ***ppd(p,q,n)***
+
+Firstly, we have two clouds of points, old (the existing points and normals in 3-D model) and new (new points and normals, what we want to integrate to the exising model)
+
+***p*** - i'th point in the new cloud of points
+
+***q*** - i'th point in the old cloud of points
+
+***n*** - i'th normal in the old cloud of normals
+
+***ppd(...)***- is the distance ergo its formula is the dot product of (difference between ***p*** and ***q***) and (***n***):
+
+<img src="https://render.githubusercontent.com/render/math?math=dot(T_{p2q}(p)-q, n)=dot((R\cdot p+t)-q,n)=[(R\cdot p+t)-q]^{T}\cdot n">
+
+***T(p)*** - rigid transform of ***p*** point, which brings it closer to the corresponding ***q*** point.
+
+<img src="https://render.githubusercontent.com/render/math?math=T_{p2q}(p) = (R \cdot  p %2B t)">
+
+Where ***R*** - rotation, ***t*** - translation.
+
+We use the Gauss-Newton method for the minimization of function.
+In the beginning, we will perform some mathematical operations:
+
+<img src="https://render.githubusercontent.com/render/math?math=E=\sum\left\|[(R\cdot p %2B t)-q]^{T}\cdot n\right\|_{2}">
+
```

Comment:
It's better to add something like this:
In Gauss-Newton method we do sequential steps by changing R and t in the direction of the function E decrease, i.e. in the direction of its gradient:
1. At each step we approximate the function E linearly as its current value plus Jacobian matrix multiplied by delta x which is concatenated delta R and delta t vectors.
1. We find delta R and delta t by solving the equation E_approx(delta_x) = 0
1. We apply delta R and delta t to current Rt transform and proceed to next iteration

To linearize E, let's approximate it in infinitesimal neighborhood.